react-graph/src/Step.tsx

```diff
@@ -0,0 +1,112 @@
+
+    /// Styles for overall div
+    const stepsStyle: React.CSSProperties = {
+        height: props.height,
+        minHeight: '210px',
+        width: props.width,
+        position: 'absolute',
+        justifyContent: 'space-evenly'
+    }
+
+    /// Styles for gray bar
+    const stepsContainerStyle: React.CSSProperties = {
+        width: '100%',
+        backgroundColor: '#DDD',
+        maxHeight: '10px',
+        height: '10%',
+        position: 'absolute',
+        top: '50%',
+        borderRadius: '10px 0 0 10px'
+    }
+
+    /// Create each circle and description
+    function getSteps() {
+        props.steps.forEach((x) => {
+
+            /// Past steps
+            if (!met && x.id !== props.activeStep) {
+                let div = <div style={{ height: '60px', width: '25px', alignContent: 'center' }}>
+                    <span style={{ height: '25px', width: '25px', backgroundColor: '#fff', borderRadius: '50%', borderWidth: '2px', borderStyle: 'solid', borderColor: '#5DC177', display: 'inline-block' }}></span>
+                    <span style={{ fontSize: '15px', fontStyle: 'italic', color: '#538897', alignContent: 'center', display: 'inline-block' }}>{x.short}</span>
+                </div>
+                divs.push(div);
+            }
+
+            /// Current active step
+            if (x.id === props.activeStep) {
+                met = true;
+                let div = <div style={{ height: '60px', width: '25px', alignContent: 'center' }}>
+                    <span className='active_circle' style={{ height: '25px', width: '25px', backgroundColor: '#fff', borderRadius: '50%', borderWidth: '2px', borderStyle: 'solid', borderColor: '#5DC177', display: 'inline-block' }}></span>
+                    <span className='active_name' style={{ fontSize: '15px', fontStyle: 'italic', color: '#538897', alignContent: 'center', display: 'inline-block' }}>{x.long}</span>
+                </div>
+                stepBar = <div style={{ width: (props.steps.indexOf(x) / (props.steps.length - 1)) * props.width, backgroundColor: '#5DC177', maxHeight: '10px', height: '10%', position: 'absolute', top: '50%', borderRadius: '10px 0 0 10px' }}></div>
+                divs.push(div);
+            }
+
+            /// Future steps
+            if (met && x.id !== props.activeStep) {
+                let div = <div style={{ height: '60px', width: '25px', alignContent: 'center' }}>
+                    <span style={{ height: '25px', width: '25px', backgroundColor: '#fff', borderRadius: '50%', borderWidth: '2px', borderStyle: 'solid', borderColor: '#D3D3D3', display: 'inline-block' }}></span>
+                    <span style={{ fontSize: '15px', fontStyle: 'italic', color: '#538897', alignContent: 'center', display: 'inline-block' }}>{x.short}</span>
+                </div>
+                divs.push(div);
+            }
+        });
+
+    }
+
+    { getSteps() }
+    return (
+        <div style={stepsStyle}>
```

Comment:
Is it possible to make the height/width arguments optional and have this fill whatever parent div it is placed in otherwise? I am not a huge fan of specifying the exact height/width. Also, these arguments might be better served as strings so one could have one such as height: '10%' or something like that.